src/golang/lib/notification/notification.go

```diff
@@ -4,25 +4,64 @@ import (
 	"context"
 
 	"github.com/aqueducthq/aqueduct/lib/collections/integration"
+	"github.com/aqueducthq/aqueduct/lib/database"
 	"github.com/aqueducthq/aqueduct/lib/lib_utils"
 	"github.com/aqueducthq/aqueduct/lib/models"
 	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/aqueducthq/aqueduct/lib/repos"
 	"github.com/aqueducthq/aqueduct/lib/vault"
 	"github.com/aqueducthq/aqueduct/lib/workflow/operator/connector/auth"
 	"github.com/dropbox/godropbox/errors"
+	"github.com/google/uuid"
 )
 
 var ErrIntegrationTypeIsNotNotification = errors.New("Integration type is not a notification.")
 
 type Notification interface {
-	// `Send()` sends a notification with `level`, and the content is `msg`.
-	//
-	// The caller always call `Send()` when a notification is generated.
-	// There could be a level preference associated with the notification integration.
-	// For example, slack and email has `level` field in config,
-	// and only notifications beyond this level will be sent.
-	// In such cases, the implementation of `Send()` should reflect the level preference.
-	Send(msg string, level shared.NotificationLevel) error
+	// `ID()` is the unique identifier, typically mapped to the integration ID.
+	ID() uuid.UUID
+
+	// `Level()` is the default level beyond which a notification should send.
```

Comment:
Can you define what "beyond which" means here? An example would be nice. Eg. what is sent for a "warning" level.